### PR TITLE
Add SubWorkflowFixtures and refactor RandomUtil

### PR DESCRIPTION
**Pull request by jmthibault79**
Developed for https://github.com/DataBiosphere/firecloud-app/issues/9 but may be more generally applicable for testing subworkflows.

**PR checklist**
- [x] For each library you've modified here, decide whether it requires a major, minor, or no version bump. (Click [here](/broadinstitute/workbench-libs/blob/develop/CONTRIBUTING.md) for guidance)

If you're doing a **major** or **minor** version bump to any libraries:
- [x] Bump the version in `project/Settings.scala` `createVersion()`
- [x] Update `CHANGELOG.md` for those libraries
- [x] I promise I used `@deprecated` instead of deleting code where possible

In all cases:
- [x] Replace the appropriate version hashes in `README.md` and the `CHANGELOG.md` for any libs you changed with `TRAVIS-REPLACE-ME` to auto-update the version string
- [ ] Get two thumbsworth of PR review
- [ ] Verify all tests go green (CI _and_ coverage tests)
- [ ] Squash commits and **merge to develop**
- [ ] Delete branch after merge

### CODE REVIEW

File: serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/fixture/BillingFixtures.scala [hunk 1]
```diff
@@ -8,27 +8,18 @@ import org.broadinstitute.dsde.workbench.model.{UserInfo, WorkbenchEmail, Workbe
...
   * billing projects of your own.  Using GPAlloc will generally be much faster, limit the creation of billing projects
   * to those tests which truly require them.
   */
-trait BillingFixtures extends ExceptionHandling with LazyLogging with CleanUp {
+trait BillingFixtures extends ExceptionHandling with LazyLogging with CleanUp with RandomUtil {
   self: TestSuite =>
 
-  // copied from WebBrowserSpec so we don't have to self-type it
-  // TODO make it common to API and browser tests
-  private def makeRandomId(length: Int = 7): String = {
```
Comment (jmthibault79) [inline serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/fixture/BillingFixtures.scala:28]: Identical to the version in `RandomUtil`

File: serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/fixture/BillingFixtures.scala [hunk 2]
```diff
@@ -59,7 +50,7 @@ trait BillingFixtures extends ExceptionHandling with LazyLogging with CleanUp {
   }
 
   private def createNewBillingProject(namePrefix: String, memberEmails: List[String] = List())(implicit token: AuthToken): String = {
-    val billingProjectName = namePrefix + "-" + makeRandomId()
+    val billingProjectName = randomIdWithPrefix(namePrefix)
```
Comment (jmthibault79) [inline serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/fixture/BillingFixtures.scala:53]: Refactored a few cases like these to be more uniform

File: serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/fixture/SubWorkflowFixtures.scala [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+
+    createPublicMethod(method)
+  }
+
+  private def methodWithSubWorkflows(workflowName: String, methodNamespace: String, child: Method, scatterCount: Int ): Method = {
+
+    // Orchestration in real environments has a globally resolvable name like "firecloud-orchestration.dsde-dev.b.o"
+    // but not in FIABs; instead they can use the docker network name
+
+    val orchUrl = if (Config.FireCloud.orchApiUrl.contains("fiab"))
```
Comment (jmthibault79) [inline serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/fixture/SubWorkflowFixtures.scala:131]: This may not be the most robust method of FIAB-detection.

File: serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/fixture/SubWorkflowFixtures.scala [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+    val entityId = "participant1"
+  }
+
+  // ---------------------------------------------------------------------------------------------------------------
+
+  private def createPublicMethod(method: Method)(implicit token: AuthToken): Method = {
+    Orchestration.methods.createMethod(method.creationAttributes)
+    // makes Method publicly readable so it can be referenced by other Methods
+    Orchestration.methods.setMethodPermissions(method.methodNamespace, method.methodName, method.snapshotId, "public", "READER")
+    register cleanUp Orchestration.methods.redact(method)
```
Comment (davidangb) [inline serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/fixture/SubWorkflowFixtures.scala]: register the cleanup before setting public, to handle the corner case where we created the method correctly but could not set permissions.

File: serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/fixture/SubWorkflowFixtures.scala (hunk 3 above)
Comment (davidangb) [reply serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/fixture/SubWorkflowFixtures.scala:131]: yeah, this feels fragile, but should work for now. I haven't checked - is there a `$RUN_CONTEXT` environment variable we can read? (or some other variable set by config/jenkins)

**Review by davidangb — APPROVED** [review]
I didn't run this myself; code looks solid, with two minor comments inline.

File: serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/fixture/SubWorkflowFixtures.scala (hunk 3 above)
Comment (jmthibault79) [reply serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/fixture/SubWorkflowFixtures.scala:131]: Since we know it at config-render time we could do something silly like like adding `fiab = true` to application.conf

**Review by aweng98 — APPROVED** [review]
LGTM
